Source/JavaScriptCore/dfg/DFGFixupPhase.cpp

```diff
@@ -4500,6 +4533,54 @@ class FixupPhase : public Phase {
         return std::tuple { arrayModes, flags };
     }
 
+    static std::optional<ArrayModes> refineArrayModesForMultiPutByVal(Node*, ArrayModes arrayModes)
+    {
+        constexpr ArrayModes supportedArrays = 0
+            | asArrayModesIgnoringTypedArrays(ArrayWithInt32)
+            | asArrayModesIgnoringTypedArrays(ArrayWithDouble)
+            | asArrayModesIgnoringTypedArrays(ArrayWithContiguous)
+            // | CopyOnWriteArrayWithInt32ArrayMode
+            // | CopyOnWriteArrayWithDoubleArrayMode
+            // | CopyOnWriteArrayWithContiguousArrayMode
+            | Int8ArrayMode
+            | Int16ArrayMode
+            | Int32ArrayMode
+            | Uint8ArrayMode
+            | Uint8ClampedArrayMode
+            // | Float16ArrayMode // Not supporting right now.
+            | Uint16ArrayMode
+            | Uint32ArrayMode
+            | Float32ArrayMode
+            | Float64ArrayMode
+            // | BigInt64ArrayMode // Not supporting right now.
+            // | BigUint64ArrayMode // Not supporting right now.
+            | 0;
+
+        if (!arrayModes)
+            return std::nullopt;
+
+        // NonArray IndexingMode includes various subtle array-like objects, including DirectArguments, String, etc.
+        // We do not support them in Multi-ByVal ops.
+        if (~supportedArrays & arrayModes)
+            return std::nullopt;
+
+        // Merge CoW ArrayModes into non-CoW ones.
+        if (arrayModes & CopyOnWriteArrayWithInt32ArrayMode)
+            arrayModes |= asArrayModesIgnoringTypedArrays(ArrayWithInt32);
+
+        if (arrayModes & CopyOnWriteArrayWithDoubleArrayMode)
+            arrayModes |= asArrayModesIgnoringTypedArrays(ArrayWithDouble);
+
+        if (arrayModes & CopyOnWriteArrayWithContiguousArrayMode)
+            arrayModes |= asArrayModesIgnoringTypedArrays(ArrayWithContiguous);
```

Comment:
I know CoW are currently not supported. Should we mask them off as well just in case they are supported later
```
if (arrayModes & CopyOnWriteArrayWithInt32ArrayMode) {
    arrayModes &= ~  CopyOnWriteArrayWithInt32ArrayMode;
    arrayModes |= asArrayModesIgnoringTypedArrays(ArrayWithInt32);
}
```

Reply:
It seems CoW arrays should just be omitted until we know what we want to do with them?